ibm/data_source_ibm_is_vpc_routing_table_routes.go

```diff
@@ -0,0 +1,167 @@
+						},
+						isRoutingTableRouteLifecycleState: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Routing Table Route Lifecycle State",
+						},
+						isRoutingTableRouteAction: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Routing Table Route Action",
+						},
+						isRoutingTableRouteDestination: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Routing Table Route Destination",
+						},
+						isRoutingTableRouteNexthop: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Routing Table Route Nexthop Address",
+						},
+						isRoutingTableRouteZoneName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Routing Table Route Zone Name",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMISVPCRoutingTableRoutesList(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	vpcID := d.Get(isRoutingTableRouteVpcID).(string)
+	routingTableID := d.Get(isRouteTableID).(string)
+	start := ""
+	allrecs := []vpcv1.Route{}
+	for {
+		listVpcRoutingTablesRoutesOptions := sess.NewListVPCRoutingTableRoutesOptions(vpcID, routingTableID)
+		if start != "" {
+			listVpcRoutingTablesRoutesOptions.Start = &start
+		}
+		result, detail, err := sess.ListVPCRoutingTableRoutes(listVpcRoutingTablesRoutesOptions)
+		if err != nil {
+			log.Printf("Error reading list of VPC Routing Table Routes:%s\n%s", err, detail)
+			return err
+		}
+		start = GetNext(result.Next)
+		allrecs = append(allrecs, result.Routes...)
+		if start == "" {
+			break
+		}
+	}
+
+	vpcRoutingTableRoutes := make([]map[string]interface{}, 0)
+
+	for _, instance := range allrecs {
+		route := map[string]interface{}{}
+		if instance.ID != nil {
+			route[isRoutingTableRouteID] = *instance.ID
+		}
+		if instance.Href != nil {
+			route[isRoutingTableRouteHref] = *instance.Href
+		}
+		if instance.Name != nil {
+			route[isRoutingTableRouteName] = *instance.Name
+		}
+		if instance.CreatedAt != nil {
+			route[isRoutingTableRouteCreatedAt] = (*instance.CreatedAt).String()
+		}
+		if instance.LifecycleState != nil {
+			route[isRoutingTableRouteLifecycleState] = *instance.LifecycleState
+		}
+		if instance.Destination != nil {
+			route[isRoutingTableRouteDestination] = *instance.Destination
+		}
+		if instance.Zone != nil && instance.Zone.Name != nil {
+			route[isRoutingTableRouteZoneName] = *instance.Zone.Name
+		}
+		if instance.NextHop != nil {
+			nexthop := *instance.NextHop.(*vpcv1.RouteNextHop)
+			if nexthop.Address != nil {
+				route[isRoutingTableRouteNexthop] = *nexthop.Address
+			}
+		}
+
+		vpcRoutingTableRoutes = append(vpcRoutingTableRoutes, route)
+	}
+	d.SetId(dataSourceIBMISVPCRoutingTableRoutesID(d))
+	d.Set(isRoutingTableRoutes, vpcRoutingTableRoutes)
```

Comment:
Please set back both isRoutingTableRouteVpcID and isRouteTableID

Reply:
Done